src/ServiceInsight/SequenceDiagram/ModelCreator.cs

```diff
@@ -25,6 +26,7 @@ public ModelCreator(List<StoredMessage> messages, IMessageCommandContainer conta
             this.container = container;
 
             Initialize();
+            OrderHandlers();
```

Comment:
What if `Initialize` created a list as a local variable and then assigned the `handlers` fields at the end, using the ordering? (The main code-smell is the reassignment of the `handlers` field.)